tests/utils.py

```diff
@@ -0,0 +1,128 @@
+import itertools
+import random
+import string
+
+import avro.schema
+
+
+def valueFromSetGenerator(possibilities=[]):
+    while True:
+        yield random.choice(possibilities)
```

Comment:
That would work if you created a generator for each test yes. I don't think it would be a problem to add the ability to do this, but I'd prefer not to block this PR on it. Would you like a TODO or issue? LMK what you think.